cmake/extensions.cmake

```diff
@@ -353,17 +398,17 @@ endmacro()
 macro(zephyr_library_named name)
   # This is a macro because we need add_library() to be executed
   # within the scope of the caller.
-  set(ZEPHYR_CURRENT_LIBRARY ${name})
-  add_library(${name} STATIC "")
+  set(ZEPHYR_CURRENT_LIBRARY ${IMAGE}${name})
+  add_library(${ZEPHYR_CURRENT_LIBRARY} STATIC "")
 
-  zephyr_append_cmake_library(${name})
+  zephyr_append_cmake_library(${ZEPHYR_CURRENT_LIBRARY})
 
-  target_link_libraries(${name} PUBLIC zephyr_interface)
+  target_link_libraries(${ZEPHYR_CURRENT_LIBRARY} PUBLIC ${IMAGE}zephyr_interface)
 endmacro()
 
 
 function(zephyr_link_interface interface)
-  target_link_libraries(${interface} INTERFACE zephyr_interface)
+  target_link_libraries(${interface} INTERFACE ${IMAGE}zephyr_interface)
```

Comment:
I'm confused about why some of these functions now prepend IMAGE (like this one), but others don't (like `zephyr_library_link_libraries`).

I feel pretty clueless about `zephyr_library_link_libraries` now; it seems to sometimes prefix IMAGE and sometimes not:

```
$ git grep -i 'zephyr_library_link_libraries.*mbedtls'
drivers/crypto/CMakeLists.txt:zephyr_library_link_libraries_ifdef(CONFIG_MBEDTLS mbedTLS)
ext/lib/crypto/mbedtls/CMakeLists.txt:  zephyr_library_link_libraries(${IMAGE}mbedTLS)
subsys/jwt/CMakeLists.txt:zephyr_library_link_libraries_ifdef(CONFIG_MBEDTLS mbedTLS)
subsys/net/ip/CMakeLists.txt:zephyr_library_link_libraries_ifdef(CONFIG_MBEDTLS mbedTLS)
subsys/net/lib/lwm2m/CMakeLists.txt:zephyr_library_link_libraries_ifdef(CONFIG_MBEDTLS mbedTLS)
```

Similarly, `subsys__bluetooth` is always prefixed with IMAGE, but libraries like CMSIS and NFFS never are.

Can you explain why this is this way? Preferably in a source code comment :). Sorry if this is just due to my not understanding `target_link_libraries` well enough.

Reply:
As far as I understood the idea zephyr_* functions should automatically prefix `${IMAGE}` to the target/variable while `target_link_libraries` is a CMake function, not a zephyr CMake function extensions. So I don't think `zephyr_library_link_libraries(${IMAGE}mbedTLS)` is correct, but it could be that some functions are not able to prepend the `${IMAGE}` variable to the name.

Reply:
> As far as I understood the idea zephyr_* functions should automatically prefix ${IMAGE} to the target/variable while target_link_libraries is a CMake function, not a zephyr CMake function extensions.

Right, but `zephyr_library_link_libraries` is not prefixing `${IMAGE}`, unlike some of the others, and some of its users do the prefixing, but others don't. I find that confusing.

Reply:
It is a bug that mbedTLS, NFFS, and CMSIS are not consistently prefixed with IMAGE, will fix.

Reply:
Fixed